module/modmanager/manager.go

```diff
@@ -126,6 +126,7 @@ func (mgr *Manager) add(ctx context.Context, conf config.Module, conn *grpc.Clie
 		name:      conf.Name,
 		exe:       conf.ExePath,
 		logLevel:  conf.LogLevel,
+		conn:      conn,
```

Comment:
Ran into a funky bug here. If a working module is reconfigured to a module that immediately crashes, `startProcess` below will error during the reconfiguration, but the module will have already been constructed here with a `nil` `conn`. The original `conn` of the module will be lost and will never be closed. We had `conn` reuse logic in `dial` before, but I think we should just instantiate the module struct with the passed in `conn` so the original `conn` is not lost in error paths.

Apologies if that was a poor explanation.. feel free to follow up offline if it didn't make sense 😆 .